Allow reverse lookups based on a character

treyhunner opened the pull request:
Sometimes I end up with a character in my clipboard and I'd like to easily lookup its name and the `\u` code I'd need to type into Python to get the same character.

This adds a reverse lookup which redirects to the correct character name URL for that purpose.

Examples:

- https://utf8.xyz/— → https://utf8.xyz/em-dash
- https://utf8.xyz/→ → https://utf8.xyz/rightwards-arrow
- https://utf8.xyz/→ → https://utf8.xyz/rightwards-arrow
- https://utf8.xyz/• → https://utf8.xyz/bullet

It might be worth considering that this endpoint could do something more useful from the perspective of `curl` (printing `\u2192` or `\N{rightwards arrow}` maybe).

sethmlarson commented on app.py:
```diff
@@ -83,6 +83,16 @@ def category_or_char(char):
         )
 
+    # Redirect single characters to their names
+    if len(char) == 1:
```
Because of grapheme clusters like flags we'd want a grapheme-aware comparison here. Maybe we need to add another library? Maybe https://pypi.org/project/grapheme?

treyhunner replied:
What would you do in that case? 🤔 Show a page listing all characters pasted and linking to each page?

That feels like a separate feature to me, mostly because I'm pretty much always looking up single-character characters. I like the idea of also handling graphemes though.

sethmlarson reviewed (commented):
Love this idea! One issue I can see:

sethmlarson commented on app.py:
```diff
@@ -83,6 +83,16 @@ def category_or_char(char):
+        except Exception:
+            pass
+        else:
+            return redirect(f"/{name}")
```
Should we use `url_for` here instead of a hard-coded path?

treyhunner replied:
Yes! I couldn't remember the name of that one and got lazy. 😅

treyhunner commented:
@sethmlarson what do you think of also adding the Python-specific `\N{...}` syntax to these pages too? I'm realizing that I usually look up characters to get their name and then use `\N{...}`, so having a quick copy-paste for that from a redirected character URL might be handy.

Somewhat related to that, if you wanted to avoid a redirect we could have two pages with the same content and use a `<link rel="canonical" href="...">` in the `<head>`. I don't have a strong feeling either way on that.

sethmlarson commented:
@treyhunner I'd be okay adding the `\N{...}` syntax!

treyhunner commented:
> @treyhunner I'd be okay adding the `\N{...}` syntax!

Added to the PR. 😊

I also added `url_for`.

Do you feel grapheme cluster support is needed before merging this?

sethmlarson commented:
Yeah we can wait on grapheme cluster support, forgot that I likely don't support those currently anyways! :shrug:
